### PR TITLE
Masterbar: add show/hide blogs to site picker

### LINKED ISSUE #2543: Masterbar: merge the "show/hide" blog list with the new site picker
The "show/hide" list of blogs in settings can be dumped and its functionality moved to the site picker.

### PR DESCRIPTION
**Pull request by nbradbury**
Fix #2543 - Added the ability to show/hide blogs from within the site picker, which replaces similar functionality that was in settings. 

Note to reviewer: ask for a demo video for an example of how this works (can't share the video here).

### CODE REVIEW

**Comment by maxme** [comment]
remove `ManageBlogsActivity` from the manifest

```
<activity
             android:name=".ui.accounts.ManageBlogsActivity"
             android:theme="@style/CalypsoTheme"/>
```

This should fix the travis build

**Comment by maxme** [comment]
I really like the new UI and the less cluttered "Account Settings" view.

Not even sure we should change that but: if a user has wpcom account and wporg site(s), we should give him the ability to hide all his wpcom sites and show only his wporg site(s).

**Comment by nbradbury** [comment]
> we should give him the ability to hide all his wpcom sites and show only his wporg site(s).

I don't think that warrants a separate feature, but there is a (slightly awkward) way to do this already. If the current site is self-hosted, go to the site picker and choose "Show/hide sites," then choose "Select all" and press Hide. This will hide all the .com sites.

**Comment by maxme** [comment]
> I don't think that warrants a separate feature, but there is a (slightly awkward) way to do this already. If the current site is self-hosted, go to the site picker and choose "Show/hide sites," then choose "Select all" and press Hide. This will hide all the .com sites.

Ahh, this is what I excepted, but one wpcom site remained. I hit a bug: https://cloudup.com/cCgK15bas2P

**Comment by nbradbury** [comment]
> Ahh, this is what I excepted, but one wpcom site remained

That's the tricky part :) I prevent the user from hiding the current site, since that would cause mayhem. This means that if a wp.com site is the current one, it won't be hidden even if you Select All > Hide.

**Comment by maxme** [comment]
> That's the tricky part :) I prevent the user from hiding the current site, since that would cause mayhem.

1. I think we should Toast a message "X wasn't hidden because it was selected" in the case the user is hiding a selected site.
2. There is still a way to cause mayhem: login with a self hosted, go to My sites -> Settings -> Remove blog.

(BTW: Should we rename "Remove blog" to "Remove site"?)

**Comment by nbradbury** [comment]
@maxme Ready for another review, thanks.

**Comment by maxme** [comment]
Looks good. Only one minor comment: if the user is not logged in wpcom, if he taps the "Add site" button, we should:

* change the "Create WordPress.com site" with "Sign in to create a wpcom site" (and open the wpcom sign in activity),
* or remove the dialog and directly open the "Add self hosted site" activity.

**Comment by nbradbury** [comment]
> or remove the dialog and directly open the "Add self hosted site" activity.

That's how it currently works already - see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2543-sitepicker-show-hide/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java#L203-L213).

File: WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java [hunk 1]
```diff
@@ -104,36 +151,64 @@ public void onEventMainThread(CoreEvents.BlogListChanged event) {
...
+            WordPress.wpDB.getDatabase().setTransactionSuccessful();
+        } finally {
+            WordPress.wpDB.getDatabase().endTransaction();
+        }
+    }
+
+    private void addSite() {
+        // if user is signed into wp.com use the dialog to enable choosing whether to
+        // create a new wp.com blog or add a self-hosted one
+        if (AccountHelper.isSignedIn()) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java]: `AccountHelper.isSignedIn()` returns true is the user is signed in wpcom *or* in a self hosted (worth a comment in AccountHelper.java).

This test should check if `AccountHelper.getDefaultAccount().isWordPressComUser()` is true.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java]: > AccountHelper.isSignedIn() returns true is the user is signed in wpcom or in a self hosted (worth a comment in AccountHelper.java).
> 
> This test should check if AccountHelper.getDefaultAccount().isWordPressComUser() is true.

TIL :) Change made in f126b8c.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**